src/main/java/org/eclipse/yasson/internal/serializer/DeserializerBuilder.java

```diff
@@ -49,13 +52,19 @@
      */
     private JsonParser.Event jsonEvent;
 
+    /**
+     * true if ordering strategy is set in associated JSONB configuration, false otherwise
+     */
+    private boolean orderStrategySet;
+    
     /**
      * Creates a new builder.
      *
      * @param jsonbContext Context.
      */
     public DeserializerBuilder(JsonbContext jsonbContext) {
         super(jsonbContext);
+        this.orderStrategySet = jsonbContext.getConfig().getProperty(JsonbConfig.PROPERTY_ORDER_STRATEGY).isPresent();
```

Comment:
we should only use `TreeMap` if the user has specified `javax.json.bind.config.PropertyOrderStrategy.LEXICOGRAPHICAL`.  If they specify `REVERSE`, `ANY`, or some other custom ordering strategy, we should continue to use `HashMap`. 
Given that, a more obvious name for the field would be `isLexOrder` or something.

Reply:
According to our email exchange, I made additional changes to deal with all ordering strategy : 

> To summarize, here are the data structures I think we should use for each:
>  - ANY: HashMap
>  - LEX: TreeMap
>  - REVERSE: ReverseTreeMap